### ISSUE TITLE
Retrieval of Jwk token is blocking

### ISSUE THREAD

**Issue by huntc** [auth0/jwks-rsa-java#147]
This could also be viewed as a feature request, but I think it is more of a bug given that I've later discovered the [following line](https://github.com/auth0/jwks-rsa-java/blob/master/src/main/java/com/auth0/jwk/JwkProvider.java#L14) of `JwkProvider` to be a blocking call:

```
    Jwk get(String keyId) throws JwkException;
```

Should we not return a future instead?

**Comment by poovamraj** [comment]
Hi @huntc, I guess we could have added this to the documentation to ensure it was conveyed properly that it is a blocking call. Returning to the future would be ideal. It would be great if you can provide a PR for this, we could help you with it if needed.

**Comment by jimmyjames** [comment]
I agree with @poovamraj we should at least update the documentation to ensure its clear. And if we can add async support in a nice way, that's something we can consider.
